container-scan/action.yaml

```diff
@@ -15,50 +27,88 @@ inputs:
   lw-access-token:
     required: true
     description: Lacework access token
+  image-platform:
+    description: Target platform to build image for (eg. linux/amd64 (default), linux/arm64, etc)
+    required: false
+    default: linux/amd64
   image-tag:
     required: true
     description: Docker image tag
+  aws-access-key-id:
+    required: false
+    default: ""
+    description: AWS access key id
+  aws-secret-access-key:
+    required: false
+    default: ""
+    description: AWS secret access key
+  aws-region:
+    required: false
+    default: "us-east-1"
+    description: AWS region
 
 outputs:
   comment-id:
     description: Comment ID of the test report
     value: ${{ steps.comment-pr.outputs.comment-id }}
   image-name:
     description: Docker image name
-    value: ${{ steps.config.outputs.image-name }}
+    value: ${{ steps.build-docker.outputs.image-name }}
 
 runs:
   using: composite
   steps:
     - name: Checkout Repository
       uses: actions/checkout@v4
 
-    - name: Extract docker image name
-      id: config
-      shell: bash
-      run: | # if docker-config file does not exist - fail
-        if [ ! -f "${{ inputs.docker-config-file }}" ]; then
-          echo "::error::Docker config file not found: ${{ inputs.docker-config-file }}" && exit 1
-        fi
-        image_name=$(jq -r .imageName ${{ inputs.docker-config-file }})
-        echo "image-name: ${image_name}"
-        echo "image-name=${image_name}" >> $GITHUB_OUTPUT
+    - name: Build docker image
+      uses: open-turo/actions-jvm/build-docker@c/add_push_load_inputs_to_build_docker
```

Comment:
This is coupling this action to just jvm services. Probably the build should be done outside of this action and this action should only receive an image tag or image id or something like that

Reply:
> This is coupling this action to just jvm services
@tagoro9 I agree with the part it gets tightly coupled with jvm services and I'll make updates to avoid that dependency. 

>  Probably the build should be done outside of this action and this action should only receive an image tag or image id or something like that
Does outside of this action means we implement build-docker composite action under `open-turo/actions-security`  something we do in 'open-turo/actions-jvm'?

Reply:
> Does outside of this action means we implement build-docker composite action under open-turo/actions-security like something we do in 'open-turo/actions-jvm'?
This does make sense as we also have to think about using this action other msvcs (python, node and others)

Reply:
@prathamesh16c and @tagoro9 as discussed I removed docker build from container-scan and added a new docker build composite and made it as optional step in container-scan action.

Tested it [here](https://github.com/turo/notifications-msvc/pull/137)

Reply:
Sounds good. I wonder if we should eventually move away from this local build, since services will already build an image for other reasons (prerelease) so we might just pass in the tag or even a local image id, but for now this works!

Reply:
> since services will already build an image for other reasons (prerelease) so we might just pass in the tag or even a local image id, but for now this works!

The container-scan action only requires that the Docker image be built, not pushed to a registry. That being said, do we actually only build images today in our CI? 🤔 The prerelease process builds and pushes the image to the registry, so I'm not sure if it's okay to push a prerelease image without scanning it first.